File: HRMS/HRMS.Web/Modules/Operations/LeavePolicy/LeavePolicyColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace HRMS.Operations.Columns;

[ColumnsScript("Operations.LeavePolicy")]
[BasedOnRow(typeof(LeavePolicyRow), CheckNames = true)]
public class LeavePolicyColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int LeavePolicyId { get; set; }
    [EditLink]
    public string PolicyName { get; set; }
    public DateTime ApplicableFromDate { get; set; }
    public string Branch { get; set; }
    public string DepartmentName { get; set; }
    public int MaxConsecutiveLeavesAllowed { get; set; }
    public ApprovalLevelMode ApprovalLevels { get; set; }
    public int PayrollCutoffDay { get; set; }
    public RecordStatus Status { get; set; }
}

File: HRMS/HRMS.Web/Modules/ServerTypes/Operations/LeavePolicyColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ApprovalLevelMode } from "./ApprovalLevelMode";
import { LeavePolicyRow } from "./LeavePolicyRow";

export interface LeavePolicyColumns {
    LeavePolicyId: Column<LeavePolicyRow>;
    PolicyName: Column<LeavePolicyRow>;
    ApplicableFromDate: Column<LeavePolicyRow>;
    Branch: Column<LeavePolicyRow>;
    DepartmentName: Column<LeavePolicyRow>;
    MaxConsecutiveLeavesAllowed: Column<LeavePolicyRow>;
    ApprovalLevels: Column<LeavePolicyRow>;
    PayrollCutoffDay: Column<LeavePolicyRow>;
    Status: Column<LeavePolicyRow>;
}

export class LeavePolicyColumns extends ColumnsBase<LeavePolicyRow> {
    static readonly columnsKey = 'Operations.LeavePolicy';
    static readonly Fields = fieldsProxy<LeavePolicyColumns>();
}

[ApprovalLevelMode]; // referenced types